### ISSUE TITLE
[Usage]: Can I get the loss of model directly?

### ISSUE THREAD

**Issue by Ther-LF** [vllm-project/vllm#9750]
Hi, great work!
I am currently optimizing LLM based on `vLLM` and need to test whether my optimizations affect the model's perplexity. Therefore, I want to obtain the model's cross-entropy loss. I have reviewed the issue: [Can I directly obtain the logits here?](https://github.com/vllm-project/vllm/issues/185) and understand that one way to get log probabilities is by setting the `logprobs` parameter in `SampleParams`. 

However, this method is not very convenient. We can only obtain the top-n most likely log probabilities for each token, and the probability of the correct token might not be among these top-n log probabilities. Setting `n` and searching for the probability of the correct token is quite cumbersome, and the cross-entropy has to be calculated manually as well. 

Therefore, I want to know if `vLLM` has a way to directly obtain cross-entropy, similar to `transformers`. 
Thank you sincerely for your help. :-)

**Comment by Skytliang** [comment]
+1 same needs

**Comment by waltonfuture** [comment]
same question. Did you solve it? Thanks.

**Comment by Wesley-Jzy** [comment]
+1 same one

**Comment by luoyuchenmlcv** [comment]
I think your proposed method (exhust the vocab table to find the logprob to train your model), is incorrect. Because cross entropy loss should be used to calculate the sequence probability by teacher forcing, which means you force the ith output token must be the one you choose. In your case of calculating cross entropy loss is without teacher forcing, and the error will drift and accumulate. Hence, if vllm cannot produce logprobs for input sequence, we cannot use this mechnism to calculate loss.
